src/robot/libraries/BuiltIn.py

```diff
@@ -1662,26 +1662,58 @@ def run_keyword_and_expect_error(self, expected_error, name, *args):
         return unic(error)
 
     @run_keyword_variant(resolve=2)
-    def repeat_keyword(self, times, name, *args):
+    def repeat_keyword(self, repeat, name, *args):
         """Executes the specified keyword multiple times.
 
         ``name`` and ``args`` define the keyword that is executed
-        similarly as with `Run Keyword`, and ``times`` specifies how many
-        the keyword should be executed. ``times`` can be given as an
+        similarly as with `Run Keyword`.
+
+        If ``repeat`` is given as count, it specifies how many the keyword
+        should be executed. ``repeat`` can be given as an
         integer or as a string that can be converted to an integer. If it is
         a string, it can have postfix ``times`` or ``x`` (case and space
         insensitive) to make the expression more explicit.
 
-        If ``times`` is zero or negative, the keyword is not executed at
+        If ``repeat`` is given as timeout, it must be in Robot Framework's
+        time format (e.g. ``1 minute``, ``2 min 3 s``) except a number like
+        42 is interpreted as count, so that a floating point number like
+        4.2 would raise exception.
+        You can use ``x`` as millisecond only with ``repeat`` including other
+        string that can be interpreted as timeout such as ``1 s 10 x``. Time
+        string only with ``x`` is interpreted as count.
+
+        If ``repeat`` is zero or negative, the keyword is not executed at
         all. This keyword fails immediately if any of the execution
         rounds fails.
 
         Examples:
         | Repeat Keyword | 5 times | Go to Previous Page |
         | Repeat Keyword | ${var}  | Some Keyword | arg1 | arg2 |
+        | Repeat Keyword | 2 mins  | Some Keyword | arg1 | arg2 |
         """
-        times = self._get_times_to_repeat(times)
-        self._run_keywords(self._yield_repeated_keywords(times, name, args))
+        try:
+            count = self._get_times_to_repeat(repeat)
+        except RuntimeError as err:
+            maxtime = self._get_maxtime_to_repeat(repeat, accept_numeric=False)
+            if maxtime is None:
+                raise err
+            keywords = self._yield_repeated_keywords_until(maxtime, name, args)
+        else:
+            keywords = self._yield_repeated_keywords(count, name, args)
+        self._run_keywords(keywords)
+
+    def _get_maxtime_to_repeat(self, timestr, accept_numeric=True):
+        if not accept_numeric:
+            if is_string(timestr) and timestr.replace('.', '', 1).isdigit():
+                return None
+            elif is_number(timestr):
+                return None
+        try:
+            timeout = timestr_to_secs(timestr)
+            maxtime = time.time() + timeout
+        except ValueError:
+            return None
+        return maxtime
```

Comment:
I would rather return `timeout` from here and let later methods calculate `maxtime`. That would allow changing the code to this:

```python
try:
    return timestr_to_secs(timestr)
except ValueError:
    return None
```